hooks/useGetBuyersChatList.tsx

```diff
@@ -0,0 +1,45 @@
+import {
+    query,
+    where,
+    FirebaseFirestoreTypes,
+    onSnapshot,
+} from "@react-native-firebase/firestore";
+import { useContext, useEffect, useState } from "react";
+import { chatsCollectionRef } from "@/constants/collections";
+import { UserContext } from "@/contexts/userContext";
+import { ChatType } from "@/types/chat";
+
+interface ParamsInterface {
+    isSeller: boolean;
+}
+
+export function useGetChatList(params: ParamsInterface) {
+    const { isSeller } = params;
+    const { user } = useContext(UserContext);
+    const [isLoading, setIsLoading] = useState(true);
+    const [chatList, setchatList] = useState<ChatType[]>([]);
+    const senderId = isSeller ? "sellerId" : "buyerId";
+
+    useEffect(() => {
+        if (!user) return;
+
+        const unsubsribe = onSnapshot(
+            query(chatsCollectionRef, where(senderId, "==", user.id)),
```

Comment:
extract the query, and add orderBy [refer](https://firebase.google.com/docs/firestore/query-data/order-limit-data#web) dateUpdated descending

```tsx
const chatQuery = query(chatsCollectionRef, where(senderId, "==", user.id), orderBy("dateUpdated", "desc"))

const unsubscribe = onSnapshot(chatQuery, ..... 
```